scandium-core/src/main/java/org/eclipse/californium/scandium/auth/PrincipalSerializer.java

```diff
@@ -146,10 +158,16 @@ private static X509CertPath deserializeCertChain(final DatagramReader reader) {
 	}
 
 	private static PreSharedKeyIdentity deserializeIdentity(final DatagramReader reader) {
-		byte[] bytes = readBytesWithLength(PSK_LENGTH_BITS, reader);
-		String virtualHost = bytes.length == 0 ? null : new String(bytes, StandardCharsets.UTF_8);
-		bytes = readBytesWithLength(PSK_LENGTH_BITS, reader);
-		return new PreSharedKeyIdentity(virtualHost, new String(bytes, StandardCharsets.UTF_8));
+		byte scoped = reader.readNextByte();
+		if (scoped == 1) {
+			byte[] bytes = readBytesWithLength(PSK_LENGTH_BITS, reader);
+			String virtualHost = bytes.length == 0 ? null : new String(bytes, StandardCharsets.UTF_8);
+			bytes = readBytesWithLength(PSK_LENGTH_BITS, reader);
+			return new PreSharedKeyIdentity(virtualHost, new String(bytes, StandardCharsets.UTF_8));
+		} else {
+			byte[] bytes = readBytesWithLength(PSK_LENGTH_BITS, reader);
+			return new PreSharedKeyIdentity(new String(bytes, StandardCharsets.UTF_8));
+		}
```

Comment:
Yes. It's a straight forward solution for the introduced "scope flag".
AFAIK this serializer is mainly used for session-tickets and some bugs are just recently fixed (last 6 months), I hope, it's not  a too big issue. Otherwise, If you would be affected, or someone else is reporting troubles, I can write to old format and append the type at the end (with default scoped).

Reply:
Personally, I'm not affected I report the issue just in case.